Source/WebCore/svg/SVGFilterElement.cpp

```diff
@@ -82,18 +82,26 @@ void SVGFilterElement::attributeChanged(const QualifiedName& name, const AtomStr
             Ref { m_primitiveUnits }->setBaseValInternal<SVGUnitTypes::SVGUnitType>(propertyValue);
         break;
     }
-    case AttributeNames::xAttr:
-        Ref { m_x }->setBaseValInternal(SVGLengthValue::construct(SVGLengthMode::Width, newValue, parseError));
+    case AttributeNames::xAttr: {
+        auto length = SVGLengthValue::constructLengthValueWithFallback(SVGLengthMode::Width, newValue, "-10%"_s);
+        Ref { m_x }->setBaseValInternal(length);
         break;
```

Comment:
```suggestion
    case AttributeNames::xAttr:
        Ref { m_x }->setBaseValInternal(SVGLengthValue::construct(SVGLengthMode::Width, newValue, "-10%"_s));
```
It's not an improvement to add a local variable, we shouldn't do that.

Reply:
Refactored and addressed this feedback. :-)